BoostRunClub/ViewModels/RunningInfoViewModel.swift

```diff
@@ -0,0 +1,149 @@
+                        value = String(format: "%d'%d\"", data.avgPace / 60, data.avgPace % 60)
+                    case .calorie:
+                        value = data.calorie <= 0 ? "--" : String(data.calorie)
+                    case .cadence:
+                        value = data.cadence <= 0 ? "--" : String(data.cadence)
+                    case .time, .interval, .bpm, .meter:
+                        return
+                    }
+                    $0.send(RunningInfo(type: $0.value.type, value: value))
+                }
+            }
+            .store(in: &cancellables)
+
+        runningService.dashBoardService.runningTime
+            .map { $0.simpleFormattedString }
+            .sink { [weak self] timeString in
+                self?.possibleTypes[.time] = timeString
+
+                self?.runningInfoSubjects.forEach {
+                    if $0.value.type == .time {
+                        $0.send(RunningInfo(type: .time, value: timeString))
+                    }
+                }
+            }.store(in: &cancellables)
+
+        runningService.runningStateSubject
+            .sink { [weak self] currentMotionType in
+                if currentMotionType == .standing {
+                    self?.showPausedRunningSignal.send()
+                    runningService.pause(autoResume: true)
+                }
+            }.store(in: &cancellables)
+    }
+
+    deinit {
+        print("[Memory \(Date())] 🌙ViewModel⭐️ \(Self.self) deallocated.")
+    }
+
+    // MARK: Inputs
+
+    func didTapPauseButton() {
+        showPausedRunningSignal.send()
+        runningService.pause(autoResume: false)
+    }
+
+    func didTapRunData(index: Int) {
+        var nextType = runningInfoSubjects[index].value.type.circularNext()
+        nextType = possibleTypes[nextType] != nil ? nextType : RunningInfoType.allCases[0]
+        runningInfoSubjects[index].send(
+            RunningInfo(
+                type: nextType,
+                value: possibleTypes[nextType, default: nextType.initialValue]
+            )
+        )
+        runningInfoTapAnimationSignal.send(index)
+    }
+
+    func viewDidAppear() {
+        if !runningService.isStarted {
+            runningService.start()
+            initialAnimationSignal.send()
+        }
+
+        if isResumed {
+            resumeAnimationSignal.send()
+        }
+    }
+
+    // MARK: Outputs
+
+    var runningInfoSubjects = [
+        RunningInfoTypeSubject(RunningInfo(type: .time)),
+        RunningInfoTypeSubject(RunningInfo(type: .pace)),
+        RunningInfoTypeSubject(RunningInfo(type: .averagePace)),
+        RunningInfoTypeSubject(RunningInfo(type: .kilometer)),
+    ]
+    var runningInfoTapAnimationSignal = PassthroughSubject<Int, Never>()
+    var initialAnimationSignal = PassthroughSubject<Void, Never>()
+    var resumeAnimationSignal = PassthroughSubject<Void, Never>()
+    var showPausedRunningSignal = PassthroughSubject<Void, Never>()
```

Comment:
```suggestion
    let runningInfoSubjects = [
        RunningInfoTypeSubject(RunningInfo(type: .time)),
        RunningInfoTypeSubject(RunningInfo(type: .pace)),
        RunningInfoTypeSubject(RunningInfo(type: .averagePace)),
        RunningInfoTypeSubject(RunningInfo(type: .kilometer)),
    ]
    let runningInfoTapAnimationSignal = PassthroughSubject<Int, Never>()
    let initialAnimationSignal = PassthroughSubject<Void, Never>()
    let resumeAnimationSignal = PassthroughSubject<Void, Never>()
    let showPausedRunningSignal = PassthroughSubject<Void, Never>()
```

프로토콜에서 var 로 되어있어서 그대로 유지하는 경우가 많은데요.
{ get } 으로만 프로퍼티 제약이 걸려있는 경우에 let 을 사용해도 충분한 상황이 있습니다.